ui/src/app/store/domain/slice.ts

```diff
@@ -38,10 +43,19 @@ const domainSlice = createSlice({
       state.saving = true;
       state.saved = false;
     },
-    setDefaultError: (state: DomainState) => {
+    setDefaultError: (
+      state: DomainState,
+      action: PayloadAction<SetDefaultErrors>
+    ) => {
       state.saving = false;
       // API seems to return the domain id in payload.error not an error message
-      state.errors = "There was an error when setting default domain.";
+      // when the domain can't be found. This override can be removed when the
+      // bug is fixed: https://bugs.launchpad.net/maas/+bug/1931654.
+      if (typeof action.payload === "number") {
```

Comment:
@ClementChaumel here I'm only overriding API errors when the error is a number so that the other errors are passed through.